### DISCUSSION TITLE
Best Practices to Export Filtered Data (filtered or not)

### DISCUSSION THREAD

**Discussion by ArthurPedroti** [TanStack/table#4684] [Q&A]
Hey everyone!

Have so many ways to handle and export data from csv for example, to do some reports and analyses that the user wants.

So, I want to know how you are handling export data for reports porpuses with TanStack Table v8 (with filters mostly times), and what experiences and insights are all having done with this implementation, to help each other with advice and tips.

**Comment by KevinVandy** [comment]
The table instance has many row models to choose from, but by default `table.getRowModel().rows` will already take pagination and filtering into account. Or you can use `table.getPrePaginationRowModel().rows`.

Here's an example in my own library built off of TanStack Table https://www.material-react-table.com/docs/examples/data-export

**Reply by ArthurPedroti** [reply]
Thanks for the tips @KevinVandy, this is a very simple and good approach, with an example easy to understand.

And a very nice initiative on trying to union theses libraries and create a "best of both worlds" library. I starred and will follow the evolution.
